fmlloader/src/main/java/net/minecraftforge/fml/loading/moddiscovery/ModFileInfo.java

```diff
@@ -65,7 +72,18 @@ public class ModFileInfo implements IModFileInfo, IConfigurable
                 .map(MavenVersionAdapter::createFromVersionSpec)
                 .orElseThrow(()->new InvalidModFileException("Missing ModLoader version in file", this));
         this.languageSpecs = new ArrayList<>(List.of(new LanguageSpec(modLoader, modLoaderVersion)));
+
         // the remaining properties are optional with sensible defaults
+        boolean notAForgeMod = false;
+        if (modLoader.equals(JAVAFML) && modLoaderVersion.containsVersion(NOT_JAVAFML_VER)) {
+            notAForgeMod = true;
+            this.properties = Map.of(NOT_A_FORGE_MOD_PROP, true);
+        } else {
+            this.properties = config.<Map<String, Object>>getConfigElement("properties")
+                    .orElse(Collections.emptyMap());
+        }
+        this.modFile.setFileProperties(this.properties);
```

Comment:
what on God's green earth is this?? if the mods.toml's language is javafml and the loader version is not `NOT_JAVAFML_VER`, we mark it as not a forge mod? would like some elaboration on this personally.

Reply:
I suck at naming stuff, especially when sleepy.

Basically, neo mods deliberately use mods.toml in the same place as Forge mods, with mostly the same format but some differences, with the `modLoader` being the same `"javafml"` but version as `2` in 1.20.4 instead of Forge's `49`... there's been a lot of controversy over neo doing this and imo it's a dick move, but it is what it is.

Reply:
how aids, but is there no better way to check for this? why not check modloader's minimum loader version to be at least 49?

Reply:
Because mods made for older Forge versions are still Forge mods ;P